tests/virt/backup_test.py

```diff
@@ -362,6 +363,90 @@ def test_start_stop_backup_engine_scratch_disks(tmpdir):
         assert drive.scratch_disk is None
 
 
+def test_start_stop_backup_engine_scratch_disks_with_scratch_ids(tmpdir):
+    vm = FakeVm()
+    dom = FakeDomainAdapter()
+
+    socket = backup.socket_path(BACKUP_1_ID)
+    scratch1 = create_scratch_disk(tmpdir, "scratch1")
+    scratch2 = create_scratch_disk(tmpdir, "scratch2")
+
+    fake_disks = create_fake_disks(vm)
+
+    # Engine sends scratch disk IDs.
+    scratch_disk_domain_uuid = make_uuid()
+    scratch_disk_image_uuid = make_uuid()
+    scratch_disk_volume_uuid = make_uuid()
+
+    # Block based scratch disk for block based disk.
+    fake_disks[0]['scratch_disk'] = {
+        'path': scratch1,
+        'type': DISK_TYPE.BLOCK,
+        'domainID': scratch_disk_domain_uuid,
+        'imageID': scratch_disk_image_uuid,
+        'volumeID': scratch_disk_volume_uuid,
+    }
+
+    # File based scratch disk for file based disk.
+    fake_disks[1]['scratch_disk'] = {
+        'path': scratch2,
+        'type': DISK_TYPE.FILE,
+        'domainID': scratch_disk_domain_uuid,
+        'imageID': scratch_disk_image_uuid,
+        'volumeID': scratch_disk_volume_uuid,
+    }
+
+    config = {
+        'backup_id': BACKUP_1_ID,
+        'disks': fake_disks
+    }
+
+    res = backup.start_backup(vm, dom, config)
+    assert dom.backing_up
+
+    backup_xml = f"""
+        <domainbackup mode='pull'>
+          <server transport='unix' socket='{socket}'/>
+          <disks>
+            <disk name='sda' backup='yes' type='block' backupmode='full'
+                exportname='sda' index='7'>
+              <driver type='qcow2'/>
+              <scratch dev='{scratch1}'>
+                <seclabel model='dac' relabel='no'/>
+              </scratch>
+            </disk>
+            <disk name='vda' backup='yes' type='file' backupmode='full'
+                exportname='vda' index='8'>
+              <driver type='qcow2'/>
+              <scratch file='{scratch2}'>
+                <seclabel model="dac" relabel="no"/>
+              </scratch>
+            </disk>
+          </disks>
+        </domainbackup>
+        """
+    assert normalized(dom.backupGetXMLDesc()) == normalized(backup_xml)
+
+    # We monitor only block based scratch disks.
+    assert vm.drives[IMAGE_1_UUID].scratch_disk == {
+        "index": 7,
+        'sd_id': scratch_disk_domain_uuid,
+        'img_id': scratch_disk_image_uuid,
+        'vol_id': scratch_disk_volume_uuid,
```

Comment:
you can get the right uuids from:

    fake_disks[0]['scratch_disk']['domainID']

Reply:
Done.